File: frontend/src/types/contracts.ts
export type RiskLevel = "low" | "medium" | "high" | "critical";

export interface Clause {
  id: string;
  type: string;
  title: string;
  description: string;
  original_text: string;
  risk_level: RiskLevel;
  recommendation: string;
  section: string;
}

export interface ContractReview {
  id: string;
  filename: string;
  file_type: string;
  uploaded_at: string;
  status: "pending" | "processing" | "completed" | "failed";
  overall_risk: RiskLevel;
  summary: string;
  clauses: Clause[];
  total_clauses: number;
  high_risk_count: number;
  medium_risk_count: number;
  low_risk_count: number;
  disclaimer: string;
}

File: backend/app/models/contracts.py
import enum
import uuid

from sqlalchemy import Enum, Float, ForeignKey, String, Text
from sqlalchemy.dialects.postgresql import JSONB, UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.models.base import Base, TimestampMixin, UUIDMixin


class ReviewStatus(str, enum.Enum):
    pending = "pending"
    processing = "processing"
    completed = "completed"
    failed = "failed"


class ClauseLibrary(Base, UUIDMixin):
    __tablename__ = "clause_library"

    clause_type: Mapped[str] = mapped_column(String(100), unique=True, nullable=False, index=True)
    name: Mapped[str] = mapped_column(String(200), nullable=False)
    description: Mapped[str] = mapped_column(Text, nullable=False)
    risk_level: Mapped[str] = mapped_column(String(20), nullable=False)  # low, medium, high
    example_text: Mapped[str] = mapped_column(Text, nullable=False)


class ContractReview(Base, UUIDMixin, TimestampMixin):
    __tablename__ = "contract_reviews"

    user_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True
    )
    filename: Mapped[str] = mapped_column(String(500), nullable=False)
    file_url: Mapped[str | None] = mapped_column(String(1000), nullable=True)
    status: Mapped[ReviewStatus] = mapped_column(Enum(ReviewStatus), default=ReviewStatus.pending, nullable=False)
    risk_score: Mapped[float | None] = mapped_column(Float, nullable=True)
    summary: Mapped[str | None] = mapped_column(Text, nullable=True)
    clauses_found: Mapped[dict | None] = mapped_column(JSONB, nullable=True)
    missing_clauses: Mapped[dict | None] = mapped_column(JSONB, nullable=True)
